### DISCUSSION TITLE
Usage of `AckKind::Nak` with WorkQueue retention policy

### DISCUSSION THREAD

**Discussion by ghost** [nats-io/nats.rs#1111] [Q&A]
Hello, we are currently using the WorkQueue policy, and I have stumbled upon an issue where `nak`ed messages are not present to the subscribers.

Our usage is basically resumed like this:
1. We have multiple worker subscribing to the same stream
2. When they start, they do:
    1. Create or get stream
    2. Create or get consumer (with same name as other workers)
    3. Listen for messages
3. When they are surcharged / blocked, we release messages in memory by `nak`ing them so they can be fetched by others workers.

I am using the WorkQueue retention policy to avoid getting messages twice since they are long running jobs.

So I have a few questions:
1. How do I ensure my message will be retried for a consumer when it has been `nak`ed?
2. Can I have multiple subscribers attached to a single consumer? I am not sure if this meant to be used like this but I haven't found another way
3. Is there a better way to do this?

**Comment by Jarema** [comment]
Hi!

Please show your stream config and consumer config.
What kind of consumer are you using? Pull or push?

**Reply by ghost** [reply]
Hi!

Sorry for the missing details, here they are.

For my stream config:

```rust
Config {
    name: self.stream_name.to_owned(),
    max_messages: 100_000,
    retention: RetentionPolicy::WorkQueue,
    ..Default::default()
}
```

For my consumer config:

```rust
Config {
    durable_name: "consumer_name".to_owned(),
    ack_wait: Duration::from_secs(300),
    max_deliver: 50,
    ..Default::default()
}
```

As for messages, I am using pull method using batch method:

```rust
let mut messages = consumer.batch().max_messages(100).messages().await?;

while let Some(Ok(message)) = messages.next().await {
    // ...
   // here I have a timeout, and if it is reached, I `nak` the rest of the messages
}
```
